src/datasource.js

```diff
@@ -246,6 +277,40 @@ export class GenericDatasource {
     return graphs;
   }
 
+  /**
+   * Transform data returned by time series query into Grafana annotation format.
+   * @param results
+   * @param options
+   * @returns Array
+   */
+  static transformAnnotationResults(results, timeField, timeEndField, textField) {
+    const annotations = [];
+    results.forEach((result) => {
+      const responseObject = {};
+      responseObject.time = Number(result[timeField]) / 1000000;
+      if (!responseObject.time && result.attributes) {
+        responseObject.time = Number(result.attributes[timeField]) / 1000000;
+      }
+      if (!responseObject.time) {
+        return; // The `.forEach` equivalent of `continue`
+      }
+
+      responseObject.text = result[textField];
+      if ((!responseObject.text || responseObject.text.length === 0) && result.attributes) {
```

Comment:
Sorry for not bringing this up earlier.  I'm just learning about JavaScript, but I think `(!responseObject.text || responseObject.text.length === 0) && result.attributes)` can be simplified to `(!responseObject.text && result.attributes)` b/c empty string is `falsy`.